### PR TITLE
Improved compatibility of multiprocessing tests with Python 3 #1950

### PR DESCRIPTION
**Pull request by Onager**
Improved compatibility of multiprocessing tests with Python 3 #1950

### CODE REVIEW

File: plaso/multi_processing/task_manager.py [hunk 1]
```diff
@@ -188,7 +190,7 @@ def _AbandonInactiveProcessingTasks(self):
           self.SampleTaskStatus(task, 'abandoned_processing')
           tasks_to_abandon.append((task_identifier, task))
 
-      for (task_identifier, task) in tasks_to_abandon:
+      for task_identifier, task in tasks_to_abandon:
```
Comment (joachimmetz) [inline plaso/multi_processing/task_manager.py]: Could you alternatively use dict.pop() inside the previous loop?

File: tests/multi_processing/task_manager.py [hunk 2]
```diff
@@ -205,6 +205,7 @@ def testUpdateLatestProcessingTime(self):
     """Tests the _UpdateLatestProcessingTime function."""
     manager = task_manager.TaskManager()
     task = manager.CreateTask(self._TEST_SESSION_IDENTIFIER)
+    task.last_processing_time = 0
```
Comment (joachimmetz) [inline tests/multi_processing/task_manager.py]: Why this change? is this a left over that could be removed?

File: tests/multi_processing/psort.py [hunk 3]
```diff
@@ -513,7 +514,8 @@ def testExportEvents(self):
 
     lines = []
     output = output_writer.ReadOutput()
-    for line in output.split(b'\n'):
+    output = codecs.decode(output, 'utf-8')
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply tests/multi_processing/psort.py]: or implement a string output writer for testing?

[the remark this answers is not in the document]
Comment (Onager) [reply tests/multi_processing/psort.py]: #1963 for that

File: tests/multi_processing/task_manager.py (hunk 2 above)
Comment (Onager) [reply tests/multi_processing/task_manager.py]: This is because the UpdateLatestProcessingTime() method compares this field against the latest_processing_time value, and Python3 errors on comparing None and int.

Comment (joachimmetz) [reply tests/multi_processing/task_manager.py]: Should we update _UpdateLatestProcessingTime instead? I assume this value being None will happen runtime as well

Comment (Onager) [reply tests/multi_processing/task_manager.py]: You requested that I not change that method ;) https://github.com/log2timeline/plaso/pull/1956/files/6b89d53e2d7c8e7089459e271c518d9e216ed853#diff-dce07e43bc91fcf6c32c3d51d2590951R254

No, this doesn't happen at runtime, because UpdateProcessingTime() is called before this method is invoked.
